Lib/datetime.py

```diff
@@ -1801,6 +1801,14 @@ def fromtimestamp(cls, timestamp, tz=None):
     @classmethod
     def utcfromtimestamp(cls, t):
         """Construct a naive UTC datetime from a POSIX timestamp."""
+        import warnings
+        warnings.warn("datetime.utcfromtimestamp() is deprecated and scheduled "
+                      "for removal in a future version. It is instead "
+                      "recommended that users use timezone-aware objects to "
+                      "represent datetimes in UTC. To get such an object from "
+                      "a timestamp, use datetime.fromtimestamp(t, datetime.UTC).",
+                      DeprecationWarning,
+                      stacklevel=2)
```

Comment:
I find this a little wordy for a warning, and tentative. The deprecated function will be removed in the future so we are not recommending an alternative, we are *telling* people to use an alternative. How about this:

```suggestion
        warnings.warn("datetime.utcfromtimestamp() is deprecated and will be "
                      "removed in a future version. Use timezone-aware objects to "
                      "represent datetimes in UTC: "
                      "datetime.fromtimestamp(t, datetime.UTC)",
                      DeprecationWarning,
                      stacklevel=2)
```

Reply:
I did think it was a bit wordy when I wrote it. My only reservation here is that users still *can* use naïve objects as they want, they just need to do `datetime.now(UTC).replace(tzinfo=None)`.

I suppose this subtle distinction is probably going to be lost on most people, so I'll just go with your wording.

Reply:
I agree that Łukasz's suggestion is an improvement. I think it is ok to suggest the slightly disruptive change; we _want_ people to use tz-aware objects[^1].

[^1]: Providing a drop-in replacement does not promote using tz-aware objects.

Reply:
> I agree that Łukasz's suggestion is an improvement. I think it is ok to suggest the slightly disruptive change; we want people to use tz-aware objects

I mean, in either case we're suggesting the change, the question is how best to communicate to users that the thing we're suggesting is not a drop-in replacement for the existing workflow. I tend to assume that when I see a deprecation warning with a code snippet in it, the code snippet is how you achieve *the same thing* in the new version. Ideally we'd make it clear that we are saying, "The thing you are doing now is not something you should continue doing, here's a new way to do it but you need to understand the implications of making this change."

That said, I'm not convinced that it's possible to convey this succinctly, so probably best to just leave it at Łukasz's wording.